### ISSUE TITLE
Adding a new file in ssh seems to fail without doing anything

### ISSUE THREAD

**Issue by miversen33** [miversen33/netman.nvim#192]
When you add a file with the neo-tree consumer in ssh, you get the following output

```
[2024-01-04 17:13:25] [SID: ufelrbgkezgfolo] [Logger: provider] [Level: TRACE]    -- /home/miversen/git/netman.nvim/lua/netman/providers/ssh.lua:callback:1353	{
  cmd_pieces = { "ssh", "-o", "ControlMaster=auto", "-o", 'ControlPath="/home/miversen/.cache/nvim/netman/tmp//%C"', "-o", "ControlPersist=10", "docker-server", "stat -L -c MODE=%f,BLOCKS=%b,BLKSIZE=%B,MTIME_SEC=%X,USER=%U,GROUP=%G,INODE=%i,PERMISSIONS=%a,SIZE=%s,TYPE=%F,NAME=%n /home/miversen/docker/stash/docker-compose.yml" },
  command = 'ssh -o ControlMaster=auto -o ControlPath="/home/miversen/.cache/nvim/netman/tmp//%C" -o ControlPersist=10 docker-server stat -L -c MODE=%f,BLOCKS=%b,BLKSIZE=%B,MTIME_SEC=%X,USER=%U,GROUP=%G,INODE=%i,PERMISSIONS=%a,SIZE=%s,TYPE=%F,NAME=%n /home/miversen/docker/stash/docker-compose.yml',
  elapsed_time = 12489860,
  elapsed_time_ml = 12.48986,
  exit_code = 1,
  opts = {
    ASYNC = true,
    EXIT_CALLBACK = <function 1>,
    STDERR_JOIN = ""
  },
  pid = 1022325,
  signal = 0,
  stderr = "stat: cannot stat '/home/miversen/docker/stash/docker-compose.yml': No such file or directory",
  stdout = {}
}
[2024-01-04 17:13:25] [SID: ufelrbgkezgfolo] [Logger: provider] [Level: WARN]     -- /home/miversen/git/netman.nvim/lua/netman/providers/ssh.lua:callback:1373	Received non-0 exit code while trying to stat ssh://docker-server///home/miversen/docker/stash/docker-compose.yml: No such file or directory {
  error = "stat: cannot stat '/home/miversen/docker/stash/docker-compose.yml': No such file or directory",
  exit_code = 1,
  locations = { "/home/miversen/docker/stash/docker-compose.yml" },
  stdout = {}
}
[2024-01-04 17:13:25] [SID: ufelrbgkezgfolo] [Logger: system] [Level: WARN]      -- /home/miversen/git/netman.nvim/lua/netman/api.lua:exit_callback:575	Provider ssh indicated a failure to connect to host of ssh://docker-server///home/miversen/docker/stash/docker-compose.yml
[2024-01-04 17:13:25] [SID: ufelrbgkezgfolo] [Logger: system] [Level: DEBUG]     -- /home/miversen/git/netman.nvim/lua/netman/api.lua:result_callback:1061	Logging message provided by table: 0x7fb6d39db108 for consumer: table: 0x7fb6d2f1d440
[2024-01-04 17:13:25] [SID: ufelrbgkezgfolo] [Logger: consumer] [Level: TRACE2]    -- ...miversen/git/netman.nvim/lua/netman/ui/neo-tree/init.lua:process_result:934	Removing process handle for ssh://docker-server///home/miversen/docker/stash/docker-compose.yml. There are currently 0 remaining processes
```

Looks like the ssh provider is failing after being unable to locate the missing file. Which of course its missing, we are asking it to create it.

**Comment by miversen33** [comment]
This does work in docker, though that is likely due to docker being synchronous vs async (which is what ssh is).

Once fixed, we will want to watch out for this in the async rewrite of docker

**Comment by miversen33** [comment]
Resolved with #196
